core/src/main/java/com/linecorp/armeria/common/stream/StreamMessage.java

```diff
@@ -676,4 +677,35 @@ default EventExecutor defaultSubscriberExecutor() {
         requireNonNull(function, "function");
         return new RecoverableStreamMessage<>(this, function, /* allowResuming */ true);
     }
+
+    /**
+     * Adapts this {@link StreamMessage} to {@link InputStream}.
+     *
+     * <p>For example:<pre>{@code
+     * StreamMessage<String> streamMessage = StreamMessage.of("foo", "bar", "baz");
+     * InputStream inputStream = streamMessage.asInputStream(x -> HttpData.wrap(x.getBytes()));
+     * byte[] expected = ImmutableList.of("foo", "bar", "baz")
+     *                                .stream()
+     *                                .map(String::getBytes)
+     *                                .reduce(Bytes::concat).get();
+     *
+     * ByteBuf result = Unpooled.buffer();
+     * int read;
+     * while ((read = inputStream.read()) != -1) {
+     *     result.writeByte(read);
+     * }
+     *
+     * int readableBytes = result.readableBytes();
+     * byte[] actual = new byte[readableBytes];
+     * for (int i = 0; i < readableBytes; i++) {
+     *     actual[i] = result.readByte();
+     * }
+     * assert Arrays.equals(actual, expected);
+     * assert inputStream.available() == 0;
+     * }</pre>
+     */
+    default InputStream asInputStream(Function<? super T, ? extends HttpData> httpDataConverter) {
```

Comment:
(nit, naming) `asInputStream` vs `toInputStream`? 🤔
```java
// StreamMessage
toDuplicator();
toInputStream();
toOutputStream();
```
Cause we already have `toDuplicator()` on StreamMessage so how about using `to*` naming?

Reply:
> Cause we already have toDuplicator() on StreamMessage so how about using to* naming?

Sounds good.

Reply:
Could we also add?
```java
default InputStream toInputStream(EventExecutor executor, Function<? super T, ? extends HttpData> httpDataConverter) {
```

Reply:
> Add default InputStream toInputStream(EventExecutor executor..)?

✅ updated.